suite/tests/CMakeLists.txt

```diff
@@ -1656,12 +1665,240 @@ function(optimize target)
+      # CMake's Unix Makefiles dependence analysis doesn't run the preprocessor
+      # and so doesn't find headers named via macros
+      # (http://www.cmake.org/Bug/view.php?id=13718)
+      set(api_ir_headers # must be full paths:
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_0args.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_1args.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_2args.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_2args_mm.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_3args.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_4args.h
+        # The following headers are x86 specific.
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_3args_avx.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_2args_avx512_evex.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_2args_avx512_vex.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_3args_avx512_evex.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_3args_avx512_evex_mask.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_3args_avx512_vex.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_4args_avx512_evex_mask_A.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_4args_avx512_evex_mask_B.h
+        ${CMAKE_CURRENT_SOURCE_DIR}/api/ir_${sfx}_5args_avx512_evex_mask.h)
+      append_property_string(SOURCE api/ir_${sfx}.c OBJECT_DEPENDS "${api_ir_headers}")
+    endif ()
+
+    if (AARCH64)
+        tobuild_api(api.ir_negative api/ir_aarch64_negative.c "" "" OFF OFF)
+    endif (AARCH64)
+  endif ()
+
+  # test static decoder library
+  tobuild_api(api.ir-static api/ir_${sfx}.c "" "" ON OFF)
+
+  if (ARM)
+    if (NOT ANDROID) # XXX i#1874: get working on Android
+      tobuild_api(api.drdecode api/drdecode_arm.c "" "" ON OFF)
+    endif ()
+  elseif (AARCH64)
+    tobuild_api(api.drdecode api/drdecode_aarch64.c "" "" ON OFF)
+  else ()
+    tobuild_api(api.drdecode api/drdecode_x86.c "" "" ON OFF)
+  endif ()
+endif ()
+
+if (DR_HOST_NOT_TARGET)
+  if (CLIENT_INTERFACE)
+    # Test postprocessing and analysis with an alternate binary directory.
+    # Limited to UNIX because the checked-in modules.log is in UNIX format
+    # (Windows needs extra fields per library).
+    # Limited to ZLIB_FOUND because we test gzipped .raw files here as well.
+    if (BUILD_CLIENTS AND UNIX AND ZLIB_FOUND)
+      # We make a writable copy so we can both write the final trace there and run our
+      # analyzer in a single command line.
+      set(srcdir
+        ${PROJECT_SOURCE_DIR}/clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw)
+      set(locdir ${PROJECT_BINARY_DIR}/drmemtrace.threadsig.aarch64)
+      file(MAKE_DIRECTORY ${locdir})
+      file(COPY ${srcdir}/raw DESTINATION ${locdir}/)
+      get_target_path_for_execution(drcachesim_path drcachesim "${location_suffix}")
+      prefix_cmd_if_necessary(drcachesim_path ON ${drcachesim_path})
+      torunonly_api(tool.drcacheoff.altbindir "${drcachesim_path}"
```

Comment:
Maybe requires further work out of scope for this PR, but can we add drdisas tests too?

Reply:
Yes, I moved the drdisas test up into the host!=target-eligible section and it passes.  Done.